instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/groovy/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2SqsTracingTest.groovy

```diff
@@ -204,4 +245,132 @@ abstract class AbstractAws2SqsTracingTest extends InstrumentationSpecification {
     resp.messages().size() == 1
     assertSqsTraces()
   }
+
+  def "batch sqs producer-consumer services: sync"() {
+    setup:
+    def builder = SqsClient.builder()
+    configureSdkClient(builder)
+    def client = builder.build()
+
+    client.createQueue(createQueueRequest)
+
+    when:
+    client.sendMessageBatch(sendMessageBatchRequest)
+
+    def resp = client.receiveMessage(receiveMessageBatchRequest)
+    def totalAttrs = resp.messages().sum {it.messageAttributes().size() }
+
+    then:
+    resp.messages().size() == 3
+    sqsAttributeInjectionEnabled || totalAttrs == 18 // No additional attributes
+    !sqsAttributeInjectionEnabled || totalAttrs == 18 + 2 // Once not injected due to too many attrs
+
+    assertTraces(xrayInjectionEnabled ? 3 : 4) {
+      trace(0, 1) {
+
+        span(0) {
+          name "Sqs.CreateQueue"
+          kind CLIENT
+        }
+      }
+      trace(1, xrayInjectionEnabled ? 4 : 3) {
+        span(0) {
+          name "Sqs.SendMessageBatch"
+          kind CLIENT // TODO: Probably this should be producer, but that would be a breaking change
```

Comment:
👍 
In general I think it'd make sense for SQS to implement the messaging semconv; but that's probably a quite large project, and we might just defer it until the messaging semconv stability.